onboard/catkin_ws/src/controls/scripts/desired_state.py

```diff
@@ -62,7 +69,9 @@ def _on_twist_received(self, twist):
         Args:
             power: ROS Twist message corresponding to desired twists in local reference frame
         """
-        self.twist = utils.parse_twist(twist)
+        parsed_twist = utils.parse_twist(twist)
+        if self.twist_state_safety(parsed_twist):
```

Comment:
I'd like this check to be done in the run loop, not the callbacks. There can be some funky behavior if we do the check in the callback. For instance, if the user is publishing a valid pose and invalid twist at the same time, the twist messages will be rejected, and the pose messages will be accepted, which will result in valid pid position control with soft estops happening at the publishing rate.

To resolve this, I think we can add the checks to lines 153 and 155:
```
elif self.twist & self.twist_state_safety(self.twist):
    # Do velocity control
elif self.power & self.twist_state_safety(self.power):
    # Do power control
```